domains/client/domain-executor/src/parent_chain.rs

```diff
@@ -15,9 +15,10 @@ use system_runtime_primitives::SystemDomainApi;
 ///
 /// - The parent chain of System Domain => Primary Chain
 /// - The parent chain of Core Domain => System Domain
-pub(crate) trait ParentChainInterface<Hash> {
-    fn head_receipt_number(&self, at: Hash) -> Result<BlockNumber, sp_api::ApiError>;
-    fn maximum_receipt_drift(&self, at: Hash) -> Result<BlockNumber, sp_api::ApiError>;
+pub(crate) trait ParentChainInterface<Block: BlockT> {
+    fn info(&self) -> Info<Block>;
```

Comment:
For this PR, only the best_hash is used in fact, perhaps `fn best_hash(&self) -> Block::Hash` is clearer to indicate what state is exactly needed.

Reply:
Make sense!